pyeudiw/satosa/default/response_handler.py

```diff
@@ -168,71 +171,115 @@ def response_endpoint(
 
         for vp_token in encoded_vps:
             # verify vp token and extract user information
-            try:
-                token_parser, token_verifier = self._vp_verifier_factory(
-                    authz_payload.presentation_submission, vp_token, request_session
-                )
-            except NotKBJWT as e400:
-                return self._handle_400(
-                    context, 
-                    "invalid vp token: not a key-bound jwt",
-                    e400
-                )
-
-            try:
-                token_issuer = token_parser.get_issuer_name()
-                whitelisted_keys = self.trust_evaluator.get_public_keys(token_issuer)
-                token_verifier.verify_signature(whitelisted_keys)
-                token_verifier.verify_challenge()
-            except MissingIssuer as e400:
-                return self._handle_400(
-                    context, 
-                    "invalid vp token: missing issuer information",
-                    e400
-                )
-            except InvalidJwkMetadataException as e500:
-                return self._handle_500(
-                    context, 
-                    "trust error: cannot fetch public keys",
-                    e500
-                )
-            except JWSVerificationError as e400:
-                return self._handle_400(
-                    context, 
-                    "invalid vp token: invalid signature",
-                    e400
-                )
-            except UnsupportedSdAlg as e400:
-                return self._handle_400(
-                    context, 
-                    "invalid vp token: unsupported signature algorithm",
-                    e400
-                )
-            except InvalidKeyBinding as e400:
-                return self._handle_400(
-                    context, 
-                    "invalid vp token: nonce or aud mismatch",
-                    e400
-                )
-            except ValueError as e400:
-                return self._handle_400(
-                    context, 
-                    "invalid vp token: missing or invalid iat claim",
-                    e400
-                )
-            except Exception as e400:
-                return self._handle_400(
-                    context, 
-                    "trust error: cannot verify vp token",
-                    e400
-                )
-
-            claims = token_parser.get_credentials()
-            iss = token_parser.get_issuer_name()
-            attributes_by_issuer[iss] = claims
-            self._log_debug(context, f"disclosed claims {claims} from issuer {iss}")
-
-        all_attributes = self._extract_all_user_attributes(attributes_by_issuer)
+            if is_sd_jwt_kb_format(vp_token):
```

Comment:
```suggestion
            if is_sd_jwt_kb_format(vp_token):
```

No, we don't need all these if statements BUT we only need to parse the presentation submission map to load the matching credential format with the matching credential presentation classes, according to the provided configuration